test/extended/kubevirt/util.go

```diff
@@ -0,0 +1,317 @@
+				},
+			},
+			Selector: labelSelector,
+		},
+	}
+	serviceClient := client.CoreV1().Services(namespace)
+	_, err := serviceClient.Create(context.Background(), service, metav1.CreateOptions{})
+	expectNoError(err)
+	expectNoError(exutil.WaitForEndpoint(client, namespace, serviceName))
+	service, err = serviceClient.Get(context.Background(), serviceName, metav1.GetOptions{})
+	expectNoError(err)
+
+	Expect(service.Spec.Ports[0].NodePort).ToNot(Equal(0))
+	return service.Spec.Ports[0].NodePort
+
+}
+
+func launchWebserverLoadBalancerService(client k8sclient.Interface, namespace, serviceName string, nodeName string, externalIPs []string) int32 {
+	labelSelector := make(map[string]string)
+	labelSelector["name"] = "web"
+	networking.CreatePodForService(client, namespace, serviceName, nodeName, labelSelector)
+
+	servicePort := 8080
+	service := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: serviceName,
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeLoadBalancer,
+			Ports: []corev1.ServicePort{
+				{
+					Protocol: corev1.ProtocolTCP,
+					Port:     int32(servicePort),
+				},
+			},
+			Selector:    labelSelector,
+			ExternalIPs: externalIPs,
+		},
+	}
+	serviceClient := client.CoreV1().Services(namespace)
+	_, err := serviceClient.Create(context.Background(), service, metav1.CreateOptions{})
+	expectNoError(err)
+	expectNoError(exutil.WaitForEndpoint(client, namespace, serviceName))
+	service, err = serviceClient.Get(context.Background(), serviceName, metav1.GetOptions{})
+	expectNoError(err)
+
+	return int32(servicePort)
+
+}
+
+func checkConnectivityToHostWithCLI(f *e2e.Framework, oc *exutil.CLI, nodeName string, podName string, host string, timeout time.Duration, hostNetwork bool) error {
+	namespace := f.Namespace.Name
+
+	e2e.Logf("Creating an exec pod on node %v", nodeName)
+	execPod := pod.CreateExecPodOrFail(f.ClientSet, namespace, fmt.Sprintf("execpod-sourceip-%s", nodeName), func(pod *corev1.Pod) {
+		pod.Spec.NodeName = nodeName
+		pod.Spec.HostNetwork = hostNetwork
+	})
+	defer func() {
+		e2e.Logf("Cleaning up the exec pod")
+		err := f.ClientSet.CoreV1().Pods(namespace).Delete(context.Background(), execPod.Name, metav1.DeleteOptions{})
+		Expect(err).NotTo(HaveOccurred())
+	}()
+
+	var stdout string
+	e2e.Logf("Waiting up to %v to wget %s", timeout, host)
+	cmd := fmt.Sprintf("wget -T 30 -qO- %s", host)
+	var err error
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(2) {
+		stdout, err = oc.Run("exec").Args(execPod.Name, "-n", namespace, "--", "/bin/sh", "-x", "-c", cmd).Output()
+		if err != nil {
+			e2e.Logf("got err: %v, retry until timeout", err)
+			continue
+		}
+		// Need to check output because wget -q might omit the error.
+		if strings.TrimSpace(stdout) == "" {
+			e2e.Logf("got empty stdout, retry until timeout")
+			continue
+		}
+		break
+	}
+	return err
+}
+
+func getSchedulableNode(serverFramework *e2e.Framework) (string, error) {
```

Comment:
`e2enode.GetRandomReadySchedulableNode()`
(which is actually better than this function because it returns a random node rather than always returning the first node, which makes for better e2e test parallelizability)